src/model/redux/userState.tsx

```diff
@@ -0,0 +1,129 @@
+import { AppActions, AppState } from './store';
+import { Dispatch } from 'react';
+import { User } from '../shared/user-interface';
+import * as UserModel from '../user-model';
+
+/*The interface for the state; how the state is supposed to look at all times.*/
+export interface UserState {
+    user: User
+    loading: boolean
+}
+
+/*Available actions to for example change the state*/
+export enum UserActionTypes {
+    ADD_USER_DATA = 'ADD_USER_DATA',
+    REMOVE_USER_DATA = 'REMOVE_USER_DATA',
+    UPDATE_USER_DATA = 'UPDATE_USER_DATA',
+    UPDATE_USER_LOADING = 'UPDATE_USER_LOADING'
+}
```

Comment:
We need another login action. I like really like the way you reuse the different actions. I think that you can do the same with the new login action